nucleus/slice.py

```diff
@@ -502,3 +528,34 @@ def check_annotations_are_in_slice(
         annotations_are_in_slice,
         reference_ids_not_found_in_slice,
     )
+
+
+def create_slice_builder_payload(
+    name: str,
+    sample_size: int,
+    sample_method: Union[str, "SliceBuilderMethods"],
+    filters: Optional["SliceBuilderFilters"],
+):
+    # enum or string
+    sampleMethod = (
+        sample_method.value
+        if isinstance(sample_method, SliceBuilderMethods)
+        else sample_method
+    )
+
+    filter_payload: Dict[str, Union[str, dict]] = {}
+    if filters is not None:
+        if filters.slice_id is not None:
+            filter_payload["sliceId"] = filters.slice_id
+        if filters.autotag is not None:
+            filter_payload["autotag"] = {
+                "autotagId": filters.autotag.autotag_id,
+                "range": filters.autotag.score_range,
+            }
+
+    return {
+        "name": name,
+        "sampleSize": sample_size,
+        "sampleMethod": sampleMethod,
+        "filters": filter_payload,
+    }
```

Comment:
I'm not a fan of using helper methods for payload building. I'd rather just use `pydantic` off the bat and serialize that. Using validators for sanitizing gives them a "home" that is guaranteed to be called for the payload.

But yeah, don't worry about refactoring this. If it works we can merge it.

Reply:
I've forgotten about pydantic. But its a good callout, would be better to have everything in one central place.
If i revisit this, i will migrate it to pydantic